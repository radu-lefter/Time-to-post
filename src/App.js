import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import GlobalStyle from './GlobalStyle';
import theme from './theme';

function App() {
  return (
    <ThemeProvider theme={theme}>
    <Router>
      <Normalize />
      <GlobalStyle />
      <Routes>
        <Route path="/search">Search</Route>
        <Route path="/">Home</Route>
        <Route path="*">404 - Not Found</Route>
      </Routes>
    </Router>
  </ThemeProvider>
  );
}

export default App;
