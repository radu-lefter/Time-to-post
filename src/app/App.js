import React from 'react';
import { Routes, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import { GlobalStyle, theme } from '../style';
import Header from '../common/header';
import Home from './Home';
import Search from './Search';


function App() {
  return (
    <ThemeProvider theme={theme}>
      <Normalize />
      <GlobalStyle />
      <Header />
      <Routes>
        <Route path="/search/javascript" element={<Search />}>Search</Route>
        <Route path="/" element={<Home />}>Home</Route>
        <Route path="*">404 - Not Found</Route>
      </Routes>
  </ThemeProvider>
  );
}

export default App;
