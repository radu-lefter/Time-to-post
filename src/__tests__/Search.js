import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../app';
import { defaultSubreddit } from '../config';

const setup = (initialPath = '/') => {

  render(
    <MemoryRouter initialEntries={[initialPath]}>
      <App />
    </MemoryRouter>,
  );

};

// test('loads top posts for subreddit in URL', async () => {
//   setup('/search/javascript');

//   screen.getByText('loading-spinner.svg');

//   // this is just a placeholder assertion that tests if the result
//   // was rendered correctly
//   expect(await screen.findByText('500')).toBeInTheDocument();
//   expect(screen.queryByText('loading-spinner.svg')).not.toBeInTheDocument();
// });

test('renders error message', async () => {
  setup('/search/failing-request');
  expect(await screen.findByText(/something went wrong/i)).toBeInTheDocument();
});


test('submitting the form updates the URL', () => {
  setup('/search/reactjs');

  const searchInput = screen.getByLabelText('r /');
  expect(searchInput.value).toBe('reactjs');

  userEvent.clear(searchInput);
  userEvent.type(searchInput, 'vuejs');
  expect(searchInput.value).toBe('vuejs');


});

test('input value changes to default subreddit when search link in header is clicked', () => {
  setup('/search/reactjs');
  const searchInput = screen.getByRole('textbox');
  const header = screen.getByRole('banner');
  const searchLink = within(header).getByRole('link', { name: /Search/ });

  userEvent.click(searchLink);

  expect(searchInput.value).toBe(defaultSubreddit);
});

async function clickFirstCellWithValue(value) {
  const heatmap = await screen.findByTestId('heatmap');
  const cell = within(heatmap).getAllByText(value)[0];
  userEvent.click(cell);
}

describe('posts table', () => {
  

  test('is not visible when cell with no posts is clicked', async () => {
    setup('/search/reactjs');
    await clickFirstCellWithValue('0');

    expect(screen.queryByRole('table')).not.toBeInTheDocument();
  });

  test('shows posts ordered by time according to cell that is clicked', async () => {
    setup('/search/reactjs');
    await clickFirstCellWithValue('4');

    const table = screen.getByRole('table');
    const tableRows = within(table)
      .getAllByRole('row')
      .slice(1);

    const tableContent = tableRows.map((row) => {
      const cells = within(row).getAllByRole('cell');
      const titleLink = within(cells[0]).getByRole('link');
      const authorLink = within(cells[4]).getByRole('link');
      return {
        title: titleLink.innerHTML,
        href: titleLink.href,
        time: cells[1].innerHTML,
        score: cells[2].innerHTML,
        numComments: cells[3].innerHTML,
        author: authorLink.innerHTML,
        authorHref: authorLink.href,
      };
    });

    expect(tableContent).toMatchSnapshot();
  });

  
});