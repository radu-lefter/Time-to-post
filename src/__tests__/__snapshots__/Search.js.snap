// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts table shows posts ordered by time according to cell that is clicked 1`] = `
Array [
  Object {
    "author": "trekhleb",
    "authorHref": "https://reddit.com/u/trekhleb",
    "href": "https://reddit.com/r/javascript/comments/ljl416/nanoneuron_7_simple_js_functions_that_explain_how/",
    "numComments": "14",
    "score": "291",
    "time": "9:03 am",
    "title": "NanoNeuron - 7 simple JS functions that explain how machines learn",
  },
  Object {
    "author": "hanifbbz",
    "authorHref": "https://reddit.com/u/hanifbbz",
    "href": "https://reddit.com/r/javascript/comments/ij9gxd/til_javascript_is_a_trademark_of_oracle/",
    "numComments": "114",
    "score": "461",
    "time": "9:26 am",
    "title": "TIL, \\"JavaScript\\" is a trademark of Oracle Corporation in the United States",
  },
  Object {
    "author": "miortiq",
    "authorHref": "https://reddit.com/u/miortiq",
    "href": "https://reddit.com/r/javascript/comments/jujhlf/askjs_how_to_write_a_clean_code/",
    "numComments": "62",
    "score": "276",
    "time": "9:44 am",
    "title": "[AskJS] How to write a clean code",
  },
  Object {
    "author": "Soupaul",
    "authorHref": "https://reddit.com/u/Soupaul",
    "href": "https://reddit.com/r/javascript/comments/iniuit/learn_the_basics_of_vanilla_js_by_building_a/",
    "numComments": "16",
    "score": "319",
    "time": "9:57 am",
    "title": "Learn the basics of Vanilla JS by building a Snake Game",
  },
]
`;
